.github/workflows/ci.yml

```diff
@@ -6,106 +6,98 @@ on:
       - master
       - develop
       - "release/**"
-env:
-  APPLITOOLS_API_KEY: ${{ secrets.APPLITOOLS_API_KEY }}
-  APPLITOOLS_BATCH_ID: ${{ github.event.pull_request.head.sha }}
 jobs:
-  prepare:
-    runs-on: ubuntu-latest
-    outputs:
-      deploy: ${{ steps.filter.outputs.deploy }}
-      vrt: ${{ steps.filter.outputs.vrt }}
-    steps:
-      - uses: dorny/paths-filter@v2
-        id: filter
-        with:
-          filters: |
-            deploy:
-              - './packages/odyssey-react/**'
-              - './packages/odyssey/src/**'
-            vrt:
-              - './packages/odyssey-react/src/**'
-              - './packages/odyssey/src/**'
-  test:
+  ci:
     runs-on: ubuntu-latest
     steps:
       - name: checkout
         uses: actions/checkout@v2
       - name: setup node
         uses: actions/setup-node@v2
+        with:
+          node-version: v12.20.0
+          cache: yarn
+      - name: node_modules cache
+        uses: actions/cache@v2
+        with:
+          path: "**/node_modules"
+          key: ${{ runner.os }}-node-modules-cache-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: ${{ runner.os }}-node-modules-cache-
       - name: install dependencies
         run: yarn install --frozen-lockfile --prefer-offline
-      - name: lint (eslint, stylelint & prettier)
+      - name: lint
         run: yarn lint
-      - name: unit tests (jest & tsc)
+      - name: type check
+        run: yarn typecheck
+      - name: unit test
         run: yarn test
-  deploy:
-    runs-on: ubuntu-latest
-    needs: [test, prepare]
-    if: ${{ needs.prepare.outputs.deploy == 'true' }}
-    steps:
-      - name: checkout
-        uses: actions/checkout@v2
+
+      - name: output change conditionals
+        uses: dorny/paths-filter@v2
+        id: changes
         with:
-          ref: ${{ github.event.pull_request.head.sha }}
-      - name: setup node
-        uses: actions/setup-node@v2
-      - name: install dependencies
-        run: yarn install --frozen-lockfile --prefer-offline
+          filters: |
+            deploy:
+              - './packages/odyssey-react/**'
+              - './packages/odyssey/src/**'
+            vrt:
+              - './packages/odyssey-react/src/**'
+              - './packages/odyssey/src/**'
+
       - name: setup AWS
+        if: steps.changes.outputs.deploy == 'true'
```

Comment:
Why add these in each job?

Reply:
There may be a slicker github actions way to do this but I'm not sure...